### ISSUE TITLE
Compatibility of generated output with copyleft free culture?

### ISSUE THREAD

**Issue by ArneBab** [CompVis/stable-diffusion#131]
I am writing free and copyleft licensed pen-and-paper roleplaying games ([1d6](https://www.1w6.org/english) ) and I would love to illustrate them with generated images (the quality is awesome!).

However since I also use artwork licensed under creativecommons attribution-sharealike and artwork licensed under the GPL, I cannot use anything that would violate the conditions of those licenses.

The [CreativeML Open RAIL-M](CreativeML Open RAIL-M) license states in section III:

> The restrictions set forth in Attachment A are considered Use-based restrictions ...  
> You are accountable for the Output you generate **and its subsequent uses**. No use of the output can contravene any provision as stated in the License.

but [cc by-sa](https://creativecommons.org/licenses/by-sa/4.0/) states:

> No additional restrictions — You may not apply legal terms or technological measures that legally restrict others from doing anything the license permits. 

and the [GPL](https://www.gnu.org/licenses/gpl) states:

> You may not impose any further restrictions on the exercise of the rights granted or affirmed under this License

Does this mean that the CreativeML Open RAIL-M license makes output from the model incompatible with copyleft Free Culture and Free Software licenses, so it would for example be illegal to use any of the output in Wikipedia?

If yes, can this be changed to make the output compatible with copyleft Free Culture and Free Software licenses?

**Comment by ArneBab** [comment]
For more discussion on the licensing, see https://news.ycombinator.com/item?id=32555208
With this kind of license, the only re-users you get are those who plan to ignore the license. All those who take the license seriously will keep away.

**Comment by Aspie96** [comment]
(I am not a lawyer and this is not legal advice).

The first thing to say is that this license is about as far away from open source as it could possibly get. Thinking of it as a license "close" to an open source one (as some admittedly do exist) is misleading. This is 100% a proprietary license (although not drafted from monetary purposes, but that's got nothing to do with it) and should be thought that way.

There is a question as to whether the text of this license could even bind someone who receives a generated image at all.

In essence: is a generated image a derivative of the model? I'd suggest absolutely not, thus someone who only receives an image wouldn't have to care about the model licensing whatsoever (since they don't need any permission to use it from the model authors), but they *might* need a license from the author of the image itself (this will vary by jurisdiction).

However, the license clarify this (in case it's not already clear by the law): "Except as set forth herein, Licensor claims no rights in the Output You generate using the Model. Since Licensor doesn't claim any right anywhere else, I don't see how someone who receives the output could possibly be bound by this license. If you use a free license (software or otherwise) for the output, and combine it with copylefted material, the end user will be bound by such free licenses.

There is a question of whether *you* will have broken the license terms of the model, but that is not an issue for those receiving the output from you.

In the case of the GPL the question is whether "source code" includes the model. If so then, clearly, nothing under the GPL can be legally mixed with output from the model (since the model itself is nonfree) without an additional permission from the author of such copylefted material. CC BY-SA has no requirements on source code and thus this question does not arise.

As for "You are accountable for the Output you generate and its subsequent uses", I am completely clueless about what it could possibly mean (accountable to whom and in what way?), except that "You" refers to the licensee of the model itself, not the output. Thus releasing the output might subject you to legal risk and to "accountability", but that's your problem, which those receiving outputs from you don't have to care about, thus it is not incompatible with copyleft.

When it comes to "No use of the output can contravene any provision as stated in the License" I don't even have a small suspicion about what it could possibly mean at all.

Of course if in a jurisdiction in which you hold rights on the output of the model you license it in a restrictive way with the purpose of complying with the model license, then the output is proprietary and cannot be mixed with anything copylefted.

Ultimately, IMHO, this model should not be used at all for absolutely anything other than inconsequential private usage, research or things that fall under fair use (or whatever similar legal doctrine exists in one's country) and don't even require to follow the license terms at all.

TL;DR: Use the model at your legal risk. If you use it on a CC BY-SA licensed work (and the output is its derivative) and you distribute it (under the CC BY-SA license), the restrictions of the model license don't apply to those receiving the output, and you complied with the CC BY-SA license. If you use it on a GPL licensed work (and the output is its derivative), then don't distribute the output. In any case you may be breaking the model license.

**Comment by ArneBab** [comment]
> When it comes to "No use of the output can contravene any provision as stated in the License" I don't even have a small suspicion about what it could possibly mean at all.
> 
> Of course if in a jurisdiction in which you hold rights on the output of the model you license it in a restrictive way with the purpose of complying with the model license, then the output is proprietary and cannot be mixed with anything copylefted.

That’s what worries me the most. Since the generated image is public domain, it may be possible for the one who ran the model to claim copyright on it, and the model license may force proprietary licensing of the results.

The legal uncertainty this creates is harrowing …

> TL;DR: Use the model at your legal risk. If you use it on a CC BY-SA licensed work (and the output is its derivative) and you distribute it (under the CC BY-SA license), the restrictions of the model license don't apply to those receiving the output, and you complied with the CC BY-SA license. If you use it on a GPL licensed work (and the output is its derivative), then don't distribute the output. In any case you may be breaking the model license.

If releasing under by-sa is safe, then you can first release under cc by-sa and then combine it with GPL works, because then the format for changing the image will be the image itself, not the model.

(that might actually be true when going directly to the GPL: if the format for making changes is the image itself, because you only run the model once and then do adaptions on the image itself, not by using the model, then the image is the source code)

**Comment by ArneBab** [comment]
Any news here?

**Comment by ArneBab** [comment]
It’s been another week — any news on this?

Is it intended, that the images can be used for almost everything _except_ for free culture?

**Comment by butameron** [comment]
I have a similar question here.
I want to release output under the CC-BY license.

> "No use of the output can contravene any provision as stated in the License.". 

A Japanese lawyer I know told me that, in general terms, releasing output under the CC-BY could be a violation of the license if such a provision exists. (* He is not familiar with this license and has not reviewed it in detail)

For example, if I release output image under the CC-BY and others used it in a manner that violates the prohibition (for example, violates "In any way that violates any applicable national, federal, state, local or international law or regulation").  In this case, I will be held legally responsible for that violation because I knew of the prohibition and did not prohibit its use.

Of course, this is a general opinion. I don't know the CreativeML Open RAIL-M license means this or not.
Therefore, I want official answer or license  for this.

Detailed questions:

* Does "You agree not to use the Model or Derivatives of the Model:" apply to use of output? 
* If not, what does "provision" of "No use of the output can contravene any provision as stated in the License." mean? (there is no provision related directly to output)
* Are there any other provisions denying to release the output under the CC-BY license and so on?

**Comment by ArneBab** [comment]
@pesser is it possible to get official answers on the questions here?

**Comment by ArneBab** [comment]
@pesser this is still open. It is an absolute blocker to my use of stable diffusion and blocks several creative projects of mine.

**Comment by TechnologyClassroom** [comment]
Regarding U.S. copyright law, the output of machine image generation is public domain and cannot be under copyright. You can read [the letter](https://fingfx.thomsonreuters.com/gfx/legaldocs/klpygnkyrpg/AI%20COPYRIGHT%20decision.pdf) regarding Zarya of the Dawn graphic novel. I am not a lawyer.

**Comment by ArneBab** [comment]
> Regarding U.S. copyright law, the output of machine image generation is public domain and cannot be under copyright. You can read [the letter](https://fingfx.thomsonreuters.com/gfx/legaldocs/klpygnkyrpg/AI%20COPYRIGHT%20decision.pdf) regarding Zarya of the Dawn graphic novel. I am not a lawyer.

This is really well-described:

> if Ms. Kashtanova were to enter those terms into an image search engine, she could not claim the images returned in response to her search were “authored” by her, no matter how similar they were to her artistic vision. — [the letter](https://fingfx.thomsonreuters.com/gfx/legaldocs/klpygnkyrpg/AI%20COPYRIGHT%20decision.pdf)

It does not answer, though, whether someone using the model would be allowed to allow all use of those images when the usage-license of the model explicitly forbids that. Since the images are public-domain that may mean that allowing the use of a model may be illegal without an explicit contract that forbids usage of the generated output in ways the model-license forbids.

So while this is an interesting ruling, it does not explicitly answer the question here.

But I am not a lawyer, so I might be wrong.

What I see is great legal uncertainty when I’m self-running the model (as opposed to using the output of a model run by others, because then the others carry the legal risk of maybe allowing me to use something that they would have been forbidden to allow; but since it’s public domain, they may not have a way to restrict others except by making me sign an explicit contract). This is too much legal uncertainty for me to include with copyleft free cultural works, because it may threaten much larger works if use of the model without restrictions is found to be illegal.
